docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,143 @@
+- TypeScript code `.ts`
+- Transpiled JavaScript code `.js`
+- Type definition files `.d.ts`
+- Source map files `.js.map` and `.d.ts.map`
+- `.npmrc`
+- `package.json`
+- `typedoc.json`
+- `tsconfig.json`
+
+Depending on which of those files you need, you can skip the generation of most of those.
+
+#### Options
+
+
+Run `generate-odata-client --help` for further options.
+
+
+|   Parameter       | Alias | Default |   Description |
+|:------------------|:-----:|:----  -:|:--------------|
+|`inputDir`| `-i` | - | This directory will be recursively searched for .edmx/.xml files. |
+|`outputDir`| `-o` | - | Directory to save the generated code in. |
+|`serviceMapping`| `-s` | `<inputDir>/service-mapping.json` | Configuration file to ensure consistent names between multiple generation runs with updated / changed metadata files. Will be generated if not existent. |
+|`forceOverwrite`| - | `false` | Exit when encountering a file that already exists. When set to true, it will be overwritten instead. Please note that compared to the --clearOutputDir option, this will not delete outdated files. |
+|`clearOutputDir`| - | `false` | Deletes EVERYTHING in the specified output directory before generating code. |
+|`generateNpmrc`| - | `true` | Generate a .npmrc file specifying a registry for @sap scoped dependencies. |
+|`generateTypedocJson`| - | `true` | Generates a typedoc.json file for each package, used for the corresponding "doc" npm script. |
+|`generatePackageJson`| - | `true` | Generate a package.json file, specifying dependencies and scripts for compiling and generating documentation. |
+|`versionInPackageJson`| - | `true?` | By default, when generating package.json file, the generator will set a version by using the generator version. It can also be set to a specific version. |
+|`generateJs`| - | `true` | Generates transpiled .js, .js.map, .d.ts and .d.ts.map files. When set to false, the generator will only generate .ts files. |
+|`generateCSN`| - | `false` | A CSN file will be generated for each service definition in the output directory. |
+
+
+### Invoke the generator programmatically
+
+You can also use the generator programmatically. You will have to provide the options.
+
+```ts
+import { generate } from '@sap-cloud-sdk/generator';
+import path from 'path';
+
+//Create your options, adapt the input & output directory
+//as well as the package name according to your setup.
+const serviceSpecsDir = path.join('test-resources', 'service-specs');
+const outputDir = "odata-client";
+
+//Create your project datastructure with all sourcefiles based on your options
+const generatorConfig = {
+  forceOverwrite: true,
+  generateJs: false,
+  useSwagger: false,
+  writeReadme: false,
+  clearOutputDir: true,
+  generateNpmrc: false,
+  generateTypedocJson: false,
+  generatePackageJson: false,
+  generateCSN: false,
+  // Unnecessary options
+  sdkAfterVersionScript: false,
+  s4hanaCloud: false
+  /*optional:
+  serviceMapping: "test/directory",
```

Comment:
Make sure to use the same quote style throughout the documentation. We should use the same style we use in our code - single quotes `'`